### PR TITLE
Migrate to C++17

### PR DESCRIPTION
**Pull request by boxanm**
Here is my attempt to migrate libpointmatcher to C++17. Unlike https://github.com/norlab-ulaval/libpointmatcher/issues/511 , I didn't encounter any errors related to gtest version. It should compile on Ubuntu18.04 and the following releases. 
Supporting Bionic is, at the moment, the only thing holding us back from going straight to C++20.

### CODE REVIEW

**Comment by simonpierredeschenes** [comment]
I will test it on my computer today, but the code looks good, good job!

**Review by simonpierredeschenes — APPROVED** [review]

**Comment by RedLeader962** [comment]
@boxanm I'm bypassing the branch protection rule since all test has passed and @simonpierredeschenes approved the the change as a code owner.  The missing check was due to a miss-configuration in the `CODEOWNERS` file that required an approval by @pomerlef  on all changes to the codebase. I'll push a fix so that he receive request for selected files that require admin oversight and keep the rest of the codebase under the responsibility of the maintainer group

**Pull request merged**
